src/crush/CrushWrapper.cc

```diff
@@ -1350,6 +1350,26 @@ int CrushWrapper::remove_rule(int ruleno)
   return 0;
 }
 
+int CrushWrapper::bucket_adjust_item_weight(CephContext *cct, crush_bucket *bucket, int item, int weight)
+{
+  if (cct->_conf->osd_crush_update_weight_set) {
+    unsigned position;
+    for (position = 0; position < bucket->size; position++)
+      if (bucket->items[position] == item)
+	break;
+    assert(position != bucket->size);
+    for (auto w : choose_args) {
+      crush_choose_arg_map arg_map = w.second;
+      crush_choose_arg *arg = &arg_map.args[-1-bucket->id];
+      for (__u32 j = 0; j < arg->weight_set_size; j++) {
+	crush_weight_set *weight_set = &arg->weight_set[j];
+	weight_set->weights[position] = weight;
```

Comment:
The ratio idea is probably correct when the weight of an item is adjusted for the first weight set in a bucket. But the ratio is going to be different  for the second (and successive) weight set. We know from experience that the second weight set is going to lower the weight of the items that have the highest target weight and vice versa. It follows that if a target weight changes and becomes higher than another, the ratio needs to be inverted. That's why I think better to just stick with the target weight until rebalancing gets a chance to run.

Am I making things more complicated than they should be ?